.travis.yml

```diff
@@ -11,6 +11,12 @@ matrix:
   - php: 7.4
     env:
     - DEPENDENCIES="--prefer-lowest --prefer-stable"
+  - php: 8.0
+    env:
+    - DEPENDENCIES="--ignore-platform-req=php"
+  - php: 8.0
+    env:
+    - DEPENDENCIES="--ignore-platform-req=php --prefer-lowest --prefer-stable"
```

Comment:
I don't think we should ignore these here. Better fix these issues properly. Not doing so makes this either not installable for normal users or it forces them to use this flag on their project too.

Reply:
Sure, makes sense, but without the flag, we cannot run the tests for php 8 until all dependencies supports php 8. What if we don't add php8 to composer.json, that way we don't officially support php8, but can still run tests. 

The tests that fail (in addition to dependencies not supporting php8 yet) are related to the vimeo/psalm version. 
* Psalm 3.11.2 is successful
* Psalm 3.18.2 fails. Seems that marking a class with @psalm-immutable conflicts with @psalm-pure somehow. Maybe this can be resolved by specifying a version constraint, but I guess this depends on how psalm develops, or removing all @psalm-pure annotations in classes annotated with @psalm-immutable